### PR TITLE
Set scalac and javac source version for all sbt subprojects

### PR DESCRIPTION
**Pull request by jkylling**
## Description
Some of the sbt subprojects are built using the source version of the java version used to compile the project.
This creates class files with versions greater than 52.0, which will fail to run on JVM 1.8, which is typically used to run Spark.
We fix this by setting the scalac and javac source version explicitly in each subproject.

## How was this patch tested?

I've tested that the updated configuration works by running
```
delta$ build/sbt clean package
delta$ find . -type f -name "*class" | xargs file -b | sort | uniq
compiled Java class data, version 52.0 (Java 1.8)
```

## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

**Comment by vkorukanti** [comment]
Thank you @jkylling for the PR. Could you please look at the failures in doc generation?

**Comment by jkylling** [comment]
Looks like this issue https://github.com/sbt/sbt/issues/355

File: build.sbt [hunk 1]
```diff
@@ -89,6 +89,8 @@ lazy val core = (project in file("core"))
     ),
 
     javaOptions += "-Xmx1024m",
+    javacOptions ++= Seq("-source", "1.8"),
+    Compile / compile / javacOptions ++= Seq("-target", "1.8"),
```
Comment (zsxwing) [inline build.sbt:93]: What's the difference between these two lines?

Could you move `scalacOptions` and `javacOptions` to `commonSettings` and remove them from each individual project?

Comment (jkylling) [reply build.sbt]: `-target 1.8` can not be passed as a parameter to `javadoc`. By restricting the option to the `Compile / compile` task `javadoc` will still work. I got the workaround from https://github.com/sbt/sbt/issues/355

Comment (jkylling) [reply build.sbt]: Thanks! I've moved the `scalacOptions` and `javacOptions` options to `commonSettings`. I took the liberty to move the `scalacOptions` specifying `target:jvm-1.8` as well.

**Review by zsxwing — APPROVED** [review]
LGTM. Thanks for the fix!

**Comment by scottsand-db** [comment]
Hi @jkylling, can you email me or dm me on delta-users slack your public email address as a Delta Lake contributor?

scott.sandre@databricks.com

**Pull request closed without merging**